packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -529,3 +529,195 @@ def _get_metrics():
         }
 
         _post_agent(kill_data)
+
+
+def test_pod_application_metrics(dcos_api_session):
+    """Launch a pod, wait for its containers to be added to the metrics service,
+    and then verify that:
+    1) Container statistics metrics are provided for the executor container
+    2) Application metrics are exposed for the task container
+    """
+    # Helper func to check for non-unique CID's in a given /containers/id endpoint
+    def check_cid(registry):
+        if len(registry) <= 1:
+            return True
+
+        cid1 = registry[len(registry) - 1]
+        cid2 = registry[len(registry) - 2]
+        if cid1 != cid2:
+            raise ValueError('{} != {}'.format(cid1, cid2))
+
+        return True
+
+    @retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
+    def test_application_metrics(agent_ip, agent_id, task_name, num_containers):
+        # Retry for two and a half minutes since the collector collects
+        # state every 2 minutes to propogate containers to the API
```

Comment:
On a point of information, the metrics pipeline collects containers every one minute. Two and a half minutes is an acceptable timeout though. Also, s/propogate/propagate/ #addingvalue